src/services/transpile.ts

```diff
@@ -48,7 +48,7 @@ namespace ts {
 
         // if jsx is specified then treat file as .tsx
         const inputFileName = transpileOptions.fileName || (transpileOptions.compilerOptions && transpileOptions.compilerOptions.jsx ? "module.tsx" : "module.ts");
-        const sourceFile = createSourceFile(inputFileName, input, options.target!); // TODO: GH#18217
+        const sourceFile = createSourceFile(inputFileName, input, getEmitScriptTarget(options));

```

Comment:
Okay, so question about transpile - usually, we look at the `package.json` to figure out if a module needs to be ESM or CJS. How do you control that with `transpileModule`?

Reply:
I’m guessing we’d either need an additional for `transpileModule` or the caller can spoof a `.mts` or `.cts` extension since the input file name need not represent a real file on disk. Is that the idea @weswigham?